spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -733,49 +830,56 @@ def set_recording(self, variable, new_state, sampling_interval=None,
             raise ConfigurationException("Variable {} is not supported".format(
                 variable))
 
-    def _get_buffered_sdram(self, vertex_slice, n_machine_time_steps):
+    def _get_buffered_sdram(
+            self, vertex_slice, n_machine_time_steps, max_rewires_per_ts):
         """
         :param ~pacman.model.graphs.commmon.Slice vertex_slice:
         :param int n_machine_time_steps:
+        :param int max_rewires_per_ts: the max possible rewires per timestep
         :rtype: list(int)
         """
         values = list()
         for variable in itertools.chain(
                 self.__sampling_rates, self.__per_timestep_variables):
             values.append(self.get_buffered_sdram(
-                variable, vertex_slice, n_machine_time_steps))
+                variable, vertex_slice, n_machine_time_steps,
+                max_rewires_per_ts))
         return values
 
     def write_neuron_recording_region(
             self, spec, neuron_recording_region, vertex_slice,
-            data_n_time_steps):
+            data_n_time_steps, max_rewires_per_ts):
         """ recording data specification
 
         :param ~data_specification.DataSpecificationGenerator spec: dsg spec
         :param int neuron_recording_region: the recording region
         :param ~pacman.model.graphs.common.Slice vertex_slice:
             the vertex slice
         :param int data_n_time_steps: how many time steps to run this time
+        :param int max_rewires_per_ts: the max possible rewires per timestep
         :rtype: None
         """
         spec.switch_write_focus(neuron_recording_region)
         spec.write_array(get_recording_header_array(
-            self._get_buffered_sdram(vertex_slice, data_n_time_steps)))
+            self._get_buffered_sdram(
+                vertex_slice, data_n_time_steps, max_rewires_per_ts)))
 
         # Write the number of variables and bitfields (ignore per-timestep)
         n_vars = len(self.__sampling_rates) - len(self.__bitfield_variables)
         spec.write_value(data=n_vars)
         spec.write_value(data=len(self.__bitfield_variables))
 
         # Write the recording data
-        recording_data = self._get_data(vertex_slice)
+        recording_data = self._get_data(vertex_slice, max_rewires_per_ts)
         spec.write_array(recording_data)
 
-    def get_buffered_sdram_per_record(self, variable, vertex_slice):
+    def get_buffered_sdram_per_record(
+            self, variable, vertex_slice, max_rewires_per_ts):
```

Comment:
I think this will end up not being called here if it isn't treated like a normal "variable".